.github/workflows/release.yml

```diff
@@ -41,12 +41,37 @@ on:
         default: true
 
 jobs:
+  check-actor-for-prod-run:
+    name: Check actor for prod run
+    if: inputs.dry_run == false
+    runs-on: ubuntu-latest
+    env:
+      ACTOR: ${{ github.actor }}
+    steps:
+    - name: Check actor for prod run
+      run: |
+        set -e
+
+        if [ "${ACTOR}" != "aws-sdk-rust-ci" ]; then
+          echo "Error: The current actor is '${ACTOR}' but only 'aws-sdk-rust-ci' is allowed to run a prod release workflow."
+          exit 1
+        fi
+
+        echo "The current actor is 'aws-sdk-rust-ci', continuing with the workflow."
+
   # We'll need to build a base image to work against if:
   # - a release was kicked off before the image build step triggered by a push to the release branch/main completed
```

Comment:
nice! in the future we should check other things as well—ensuring permission to publish to crates.io, etc.